sormas-backend/src/main/java/de/symeda/sormas/backend/caze/CaseService.java

```diff
@@ -1048,6 +1054,24 @@ public void deletePermanent(Case caze) {
 		super.deletePermanent(caze);
 	}
 
+	private Subquery<Long> getContactVisitsSubqueryForDelete(CriteriaBuilder cb, CriteriaDelete<Visit> cd, Root<Visit> visitRoot) {
+		Subquery<Long> contactVisitsSubquery = cd.subquery(Long.class);
+		Root<Visit> subqueryRoot = contactVisitsSubquery.from(Visit.class);
+		Join<Visit, Contact> visitContactJoin = subqueryRoot.join(Visit.CONTACTS, JoinType.INNER);
+		contactVisitsSubquery.where(cb.equal(subqueryRoot.get(Visit.ID), visitRoot.get(Visit.ID)));
+		contactVisitsSubquery.select(visitContactJoin.get(Visit.ID));
+		return contactVisitsSubquery;
+	}
+
+	private Subquery<Long> getContactVisitsSubqueryForUpdate(CriteriaBuilder cb, CriteriaUpdate<Visit> cu, Root<Visit> visitRoot) {
```

Comment:
If I'm not totally mistaken, those two methods are identical, so please get rid of one of them and give the other one a more generic name

Reply:
remove one of the methods